mne/channels/tests/test_montage.py

```diff
@@ -1304,6 +1304,75 @@ def test_set_montage_with_missing_coordinates():
     )
 
 
+def test_get_montage():
+    """Test get montage from Instance.
+
+    Test with standard montage and then loaded in montage.
+    """
+    # 1. read in testing data and assert montage roundtrip
+    # for testing dataset: 'test_raw.fif'
+    raw = read_raw_fif(fif_fname)
+    raw = raw.rename_channels(lambda name: name.replace('EEG ', 'EEG'))
+    raw2 = raw.copy()
+    # get montage and then set montage and
+    # it should be the same
+    montage = raw.get_montage()
+    raw.set_montage(montage, on_missing='ignore')
+    assert_object_equal(raw.info['chs'], raw2.info['chs'])
```

Comment:
can you test dig too?

Reply:
So I'm not sure why this is the case for this test dataset, but `info['dig']` contains an extra `EEG #0` digPoint, whereas when we do the set_montage that goes away since there is no `EEG000` or `REF` channel inside the actual `info['ch_names']`.

This results in 145 digpoints after set_montage, but 146 digpoints before. This occurs at index 85. Everything else matches up. I confirmed that this is the Reference DigPoint because it has the exact same positions as the reference location in all the `info['chs'][ii]['loc'][3:6]`.

**Possible solutions?**
1. Fix when I call `set_montage`: I need to keep track of the fact that there might be a REF location set in the raw file, but no actual REF channel in the dataset by name (EEG000, REF), and if so, add that DigPoint to the `info['dig']. The issue now is when I assert objects are equal, the order is not the same as before I do `set_montage()`

2. Fix when I call `get_montage`: Here, it's not clear to me how I would do this, but I would need a way of making a montage that makes sure to set the dig later with an extra DigPoint even if there is no channel name present.

3. If I don't compare the dig, then everything else matches. That is `info['chs']` because the REF location is encoded into the [3:6] inside the `'loc'` dictionary.

Reply:
Sorry if this is not the right interpretation (I'm not 100% sure I fully understand Digs vs channel locations yet, so I might be missing something). I did a bit of debugging and pushed up the updated unit test so you can see where it breaks.